packages/core/useStorage/index.ts

```diff
@@ -181,29 +181,38 @@ export function useStorage<T extends (string | number | boolean | object | null)
 
   watch(keyComputed, () => update(), { flush })
 
+  let firstMounted = false
+  const onStorageEvent = (ev: StorageEvent): void => {
+    if (initOnMounted && !firstMounted) {
+      return
+    }
+
+    update(ev)
+  }
+  const onStorageCustomEvent = (ev: CustomEvent<StorageEventLike>): void => {
+    if (initOnMounted && !firstMounted) {
+      return
+    }
+
+    updateFromCustomEvent(ev)
+  }
+
   if (window && listenToStorageChanges) {
-    tryOnMounted(() => {
-      /**
-       * Attaching event listeners here should be fine since we are in a mounted hook
-       *
-       * The custom event is needed for same-document syncing when using custom
-       * storage backends, but it doesn't work across different documents.
-       *
-       * TODO: Consider implementing a BroadcastChannel-based solution that fixes this.
-       */
-      if (storage instanceof Storage)
-        useEventListener(window, 'storage', update, { passive: true })
-      else
-        useEventListener(window, customStorageEventName, updateFromCustomEvent)
+    if (storage instanceof Storage)
+      useEventListener(window, 'storage', onStorageEvent, { passive: true })
+    else
+      useEventListener(window, customStorageEventName, onStorageCustomEvent)
+  }
 
-      if (initOnMounted)
-        update()
+  if (initOnMounted) {
+    tryOnMounted(() => {
+      firstMounted = true
+      update()
     })
```

Comment:
This is needed for SSR maybe? Why the hooks need to be added on mount in the first place?

Reply:
i'm just maintaining whatever was there before

there's an `initOnMounted` option, that's why this is a thing. don't know why that is an option though